scripts/go_gen_p4_const.go

```diff
@@ -0,0 +1,171 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/ettle/strcase"
+	"github.com/golang/protobuf/proto"
+	p4ConfigV1 "github.com/p4lang/p4runtime/go/p4/config/v1"
+	"github.com/pborman/getopt/v2"
+)
+
+const (
+	P4INFO_PATH = "conf/p4/bin/p4info.txt"
+
+	COPYRIGHT_HEADER = "/*\n* Copyright 2022-present Open Networking Foundation\n*\n* SPDX-License-Identifier: Apache-2.0\n*/\n"
+
+	CONST_OPEN  = "//noinspection GoSnakeCaseUsage\nconst (\n"
+	CONST_CLOSE = `)`
+
+	UINT32_STRING = "uint32 = "
+
+	HF_VAR_PREFIX         = "Hdr_"
+	TBL_VAR_PREFIX        = "Table_"
+	CTR_VAR_PREFIX        = "Counter_"
+	DIRCTR_VAR_PREFIX     = "DirectCounter_"
+	ACT_VAR_PREFIX        = "Action_"
+	ACTPARAM_VAR_PREFIX   = "ActionParam_"
+	ACTPROF_VAR_PREFIX    = "ActionProfile_"
+	PACKETMETA_VAR_PREFIX = "PacketMeta_"
+	MTR_VAR_PREFIX        = "Meter_"
+)
+
+func generate(p4info *p4ConfigV1.P4Info, builder *strings.Builder) *strings.Builder {
+	//HeaderField IDs
+	builder.WriteString("//HeaderFields\n")
+	for _, element := range p4info.GetTables() {
+		for _, matchField := range element.MatchFields {
+			tableName := element.GetPreamble().GetName()
+			name, ID := matchField.GetName(), strconv.FormatUint(uint64(matchField.GetId()), 10)
+			name = strcase.ToPascal(name)
+
+			builder.WriteString(HF_VAR_PREFIX + tableName + name + "\t" + UINT32_STRING + ID + "\n")
+		}
+	}
+	// Tables
+	builder.WriteString("//Tables\n")
+	for _, element := range p4info.GetTables() {
+		name, ID := element.GetPreamble().GetName(), strconv.FormatUint(uint64(element.GetPreamble().GetId()), 10)
+		name = strcase.ToPascal(name)
+
+		builder.WriteString(TBL_VAR_PREFIX + name + "\t" + UINT32_STRING + ID + "\n")
+	}
+	// Actions
+	builder.WriteString("//Actions\n")
+	for _, element := range p4info.GetActions() {
+		name, ID := element.GetPreamble().GetName(), strconv.FormatUint(uint64(element.GetPreamble().GetId()), 10)
+		name = strcase.ToPascal(name)
+
+		builder.WriteString(ACT_VAR_PREFIX + name + "\t" + UINT32_STRING + ID + "\n")
+	}
+	// Indirect Counters
+	builder.WriteString("//IndirectCounters\n")
+	for _, element := range p4info.GetCounters() {
+		name, ID := element.GetPreamble().GetName(), strconv.FormatUint(uint64(element.GetPreamble().GetId()), 10)
+		name = strcase.ToPascal(name)
+
+		builder.WriteString(CTR_VAR_PREFIX + name + "\t" + UINT32_STRING + ID + "\n")
+	}
+	// Direct Counters
+	builder.WriteString("\t//DirectCounters\n")
```

Comment:
```suggestion
	builder.WriteString("//DirectCounters\n")
```
make consistent with other comments.